addons/sale/wizard/sale_make_invoice_advance.py

```diff
@@ -109,13 +118,37 @@ def _compute_product_id(self):
             if wizard.count == 1:
                 wizard.product_id = wizard.company_id.sale_down_payment_product_id
 
-    @api.depends('sale_order_ids', 'advance_payment_method', 'deduct_down_payments')
+    @api.depends('amount', 'fixed_amount', 'advance_payment_method', 'amount_to_invoice')
+    def _compute_display_invoice_amount_warning(self):
+        for wizard in self:
+            invoice_amount = wizard.fixed_amount
+            if wizard.advance_payment_method == 'percentage':
+                invoice_amount = wizard.amount / 100 * wizard.sale_order_ids.amount_total
+            wizard.display_invoice_amount_warning = invoice_amount > wizard.amount_to_invoice
+
+    @api.depends('sale_order_ids')
     def _compute_display_draft_invoice_warning(self):
         for wizard in self:
-            wizard.display_draft_invoice_warning = (
-                wizard.advance_payment_method == 'delivered' and
-                any(aml.parent_state == 'draft' for aml in wizard.sale_order_ids.order_line.invoice_lines)
-            )
+            wizard.display_draft_invoice_warning = wizard.sale_order_ids.invoice_ids.filtered(lambda invoice: invoice.state == 'draft')
```

Comment:
What about the `delivered` part? Is this wanted?

Reply:
Yes, the conditions to display the warning (and the warning itself) have changed.
We now want to show it as soon as there is at least one draft invoice.